tests/foreman/ui/test_ldap_authentication.py

```diff
@@ -1234,3 +1236,44 @@ def test_email_of_the_user_should_be_copied(session, auth_source_ipa, ipa_data,
     with session:
         user_value = session.user.read(ipa_data['user_ipa'])
         assert user_value['user']['mail'] == result
+
+
+@tier2
+def test_deleted_idm_user_should_not_be_able_to_login(auth_source_ipa, ldap_tear_down):
+    """After deleting a user in IDM, user should not be able to login into satellite
+
+    :id: 18ad0526-e083-11ea-b1ad-4ceb42ab8dbc
+
+    :steps:
+        1. Create a new auth source with onthefly enabled
+        2. Create a new user in IDM and assigning a group to it.
+        3. Login to satellite to create the user
+        4. Delete the user from IDM
+        5. Try login to the satellite from the user
+
+    :expectedresults: User login fails
+    """
+    run_command(
+        cmd=f"echo {settings.ipa.password_ipa} | kinit admin", hostname=settings.ipa.hostname_ipa
+    )
+    test_user = gen_string('alpha')
+    run_command(
+        cmd=f"echo {settings.ipa.password_ipa} | ipa user-add {test_user} --first={test_user} "
+        f"--last={test_user} --password",
+        hostname=settings.ipa.hostname_ipa,
+    )
+    group = settings.ipa.grpbasedn_ipa.split(',')
+    for line in group:
+        if 'group' in line:
+            _, group = line.split('=')
+            break
+    run_command(
+        cmd=f"ipa group-add-member {group} --user={test_user}", hostname=settings.ipa.hostname_ipa,
+    )
+    with Session(user=test_user, password=settings.ipa.password_ipa) as ldapsession:
+        ldapsession.task.read_all()
```

Comment:
Do we have to start entire UI session for this?
Do we have similar setup in some other test? So here we could just assume login works, because if it doesn't, some other test would tell us?

In `robottelo/helpers.py` there's `get_web_session()` function that sends login data through HTTP library directly. This way you will skip setup and teardown of browser, which I believe are quite costly. Could you try using this helper? Will it have reasonable impact of performance (i.e. if we save 5 seconds, maybe it's not worth it; if we save 15, I would definitely consider it).

Reply:
Thankyou for informing about the get_web_session method, I was not aware of that. The thing is I want the newly created user to try login in the satellite. In get_web_session(), it logs in as admin.